openmc/mgxs/library.py

```diff
@@ -717,3 +725,307 @@ def load_from_file(filename='mgxs', directory='mgxs'):
+                    if ((('scatter matrix' in self.mgxs_types) and
+                         ('nu-scatter matrix' in self.mgxs_types))):
+                        xsdata.set_multiplicity(
+                            self.all_mgxs[id]['nu-scatter matrix'],
+                            self.all_mgxs[id]['scatter matrix'],
+                            xs_type=xs_type, subdomains=(id,),
+                            nuclides=[nuclide])
+                        xsdata.multiplicity = \
+                            np.nan_to_num(xsdata.multiplicity)
+                        using_multiplicity = True
+                    else:
+                        using_multiplicity = False
+
+                    if using_multiplicity:
+                        xsdata.set_scatter(
+                            self.all_mgxs[id]['nu-scatter matrix'],
+                            xs_type=xs_type, subdomains=(id,),
+                            nuclides=[nuclide])
+                    else:
+                        if 'nu-scatter matrix' in self.mgxs_types:
+                            xsdata.set_scatter(
+                                self.all_mgxs[id]['nu-scatter matrix'],
+                                xs_type=xs_type, subdomains=(id,),
+                                nuclides=[nuclide])
+                            # Since we are not using multiplicity, then
+                            # scattering multiplication (nu-scatter) must be
+                            # accounted for approximately by using an adjusted
+                            # absorption cross section.
+                            if 'total' in self.mgxs_types:
+                                xsdata.absorption = \
+                                    np.subtract(xsdata.total,
+                                                np.sum(xsdata.scatter[0, :, :],
+                                                       axis=1))
+                            else:
+                                msg = "Absorption cross section must be " + \
+                                      "provided if using a transport cross" + \
+                                      " section and while not providing a " + \
+                                      "scattering matrix"
+                                raise ValueError(msg)
+                        else:
+                            msg = "No nu-scatter matrix data was provided. " +\
+                                  "This means neutron balance cannot be " + \
+                                  "achieved since (n,xn) multiplication is " +\
+                                  "ignored."
+                            warn(msg)
+                            xsdata.set_scatter(
+                                self.all_mgxs[id]['scatter matrix'],
+                                xs_type=xs_type,
+                                subdomains=(id,),
+                                nuclides=[nuclide])
+
+                    xsdatas.append(xsdata)
```

Comment:
Alright so I think I've digested what is going on here. It seems to me that there are two largely redundant code blocks for case with/without MGXS tallied by nuclide. I am curious if we can improve code re-use by taking advantage of the fact that even MGXS which are not tallied by nuclide have a `'total'` "nuclide" (see lines [553-558](https://github.com/mit-crpg/openmc/blob/develop/openmc/mgxs/mgxs.py#L553) of mgxs.py).

@nelsonag do you think it would be possible to use the second code block defined for MGXS tallied by nuclide, and apply it to those which are tallied for `'total'` "nuclides" as well?